### DISCUSSION TITLE
Problem with plotMap_variantPairs function

### DISCUSSION THREAD

**Discussion by pj-decastro** [ORTools4MTEX/ORTools#70] [Q&A] [answered]
Dear developers,

I am experiencing a problem with the function plotMap_variantPairs. It seems to be restricted to when I enable the 'parentGrainId' input, since it works fine with the whole EBSD map. This is the following error message:
```matlab
Error using reshape
Number of elements must not change. Use [] as one of the size inputs to automatically calculate
the appropriate size for that dimension.
Error in computeVariantGrains (line 34)
isIndexed_L = reshape(isIndexed,l_ebsd,1);
Error in plotMap_variantPairs (line 28)
[grains,ebsd] = computeVariantGrains(job,'parentGrainId',pGrainId);
```
It works fine when I use the MTEX Martensite dataset as in ORTools tutorial 7. 

Thank you in advance,
Pedro

**Comment by AzdiarGazder** [comment]
Hi @pj-decastro,

I checked [ORTools_example07](https://github.com/ORTools4MTEX/ORTools/blob/develop/ORTools_example07.m).
However, like you, I could not replicate the error at all.

Please upload your script and map.
It will help out with diagnosing the problem, if any.

<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your assistance in this regard enables us to close discussions as complete. **</ins>

Warm regards,
Azdi

**Reply by pj-decastro** [reply]
Hello Azdi,

I will send both documents via e-mail. Hope that is not a problem.

Thank you,
Pedro

**Reply by AzdiarGazder** [reply]
Hi @pj-decastro,

Thank you for sending me your *.ang file via email.

The bug originates in MTEX (specifically for hexagonal grid map data) and subsequently affects ORTools.
Given that the MTEX workshop is on this week, it will be a little while before the maintainers implement all changes into the developer version.

So, for now, please follow the steps outlined below to help mitigate this issue.

1. Please download the latest developer version of [MTEX](https://github.com/mtex-toolbox/mtex).
There have been some overnight back-end implementations that are needed.

2. Please download the latest developer version of [ORTools](https://github.com/ORTools4MTEX/ORTools).
Minor modifications have been implemented specifically to use hexagonal grid maps without error.

3. Please extract the 3 files contained in this *.ZIP file [mtex_EBSDAnalysis_@EBSD_private.zip](https://github.com/ORTools4MTEX/ORTools/files/14579368/mtex_EBSDAnalysis_%40EBSD_private.zip) and paste them into the following folder of your MTEX installation: 
...mtex_EBSDAnalysis_@EBSD_private (replace "_" with "\" for a valid path)

4.  Please extract the 2 files contained in this *.ZIP file [sampleScripts_forPdC.zip](https://github.com/ORTools4MTEX/ORTools/files/14565615/sampleScripts_forPdC.zip) and paste them into the root folder of ORTools (i.e. - the one containing the example scripts).
      (a) The first script is called "ORTools_forPdC1_peakFit.m".
      Since you have both parent and child orientations in your map, you can use the peak-fit functionality in ORTools to find the OR that is the best starting guess OR.
      
      When I ran the script, I put the horizontal threshold at X = 41.7 degrees; Y =  0.0281 to isolate the highest peak.
      The last figure of this script shows that best starting guess OR for your data is the inverse Greninger-Troiano OR.
      This is because the iGT OR has the lowest disorientation from the map data (in Fig. 1 the maroon plot is closest and/or partially overlaps the blue plot).
      
      (b) The second script is called "ORTools_forPdC2_example07Modified.m".
      Here I used the iGT OR as the starting OR to reconstruct the PAGs without error.


<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your assistance in this regard enables us to close discussions as complete. **</ins>

Hope this helps.

Warm regards,
Azdi

**Reply by frankNiessen** [reply, marked as answer]
Hi @pj-decastro,

discard the above patch provided by Azdiar and download the latest developer version of ORTools. I have now updated `computeVariantGrains` to not rely on the `gridify` command in MTEX which was at the root of the problem. 

Best wishes
Frank
